build.gradle

```diff
@@ -0,0 +1,331 @@
+			"auth2/kbase-auth2test-0.2.4.jar",
+			"common/$kbaseCommonJar",
+			"handle/AbstractHandleClient-1.0.0.jar",
+			"kidl/kbase-kidl-parser-1409261812-7863aef.jar",
+			"sample/SampleServiceClient-0.1.1.jar",
+			"shock/shock-client-0.1.0.jar"
+		]
+		jars.each { jar ->
+			def jarName = jar.split("/")[-1]
+			def file = file("$downloadDir/$jarName")
+			if (!file.exists()) {
+				ant.get(src: "$url/$jarName", dest: "$downloadDir/$jarName")
+			}
+		}
+	}
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDir src
+			exclude "**/test/**"
+		}
+	}
+	test {
+		java {
+			srcDir src
+		}
+		resources {
+			srcDir src
+			include "**/*.properties"
+			include "**/*.cfg"
+			include "**/*.spec"
+			include "**/*.instance.*"
+			include "**/instance.*"
+			include "**/*.instance"
+			include "**/*.html"
+			include "**/*.css"
+			include "**/*.gif"
+			include "**/*.js"
+			include "**/*.png"
+			include "**/*.txt"
+			include "**/*.weirdsuffix"
+		}
+	}
+}
+
+tasks.withType(JavaCompile) {
+	classpath = configurations.allDeps
+	options.compilerArgs = ["-source", "11", "-target", "11", "-g"]
+}
+
+task clientJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	include "us/kbase/workspace/*.class"
+	exclude "us/kbase/workspace/WorkspaceServer.class"
+	include "us/kbase/common/service/*.class"
+	exclude "us/kbase/common/service/JsonServer*"
+	archiveFileName = clientJarFile
+	destinationDirectory = file("$dist/client")
+}
+
+task serviceJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	archiveFileName = serviceJarFile
+	destinationDirectory = file("$dist")
+}
+
+task buildJars {
+	dependsOn serviceJar, clientJar
+}
+
+task unzipJar(type: Copy) {
+	from zipTree("$jarDir/kbase/common/$kbaseCommonJar")
+	into "$tempunpack"
+}
+
+// by default, the files go into build/docs/javadoc
+tasks.javadoc {
+	dependsOn unzipJar
+	classpath = configurations.allDeps
+	failOnError = false
+	options {
+		author = false
+		noDeprecated = false
+		noDeprecatedList = false
+		noIndex = false
+		noNavBar = false
+		noTree = false
+		splitIndex = true
+		use = true
+		version = true
+		links = [
+			"http://download.oracle.com/javase/11/docs/api/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+		]
+	}
+	source = fileTree(dir: "$src/us/kbase",
+	include: [
+		"workspace/*.java",
+		"common/service/*.java"
+	],
+	exclude: [
+		"workspace/WorkspaceServer.java",
+		"common/service/JsonServer*",
+		"common/service/JacksonTupleModule.java",
+		"common/service/JsonClientCaller.java",
+		"common/service/JsonTreeTraversingParser.java",
+		"common/service/KBaseJsonParser.java"
+	])
+	source += fileTree(dir: "$tempunpack/us/kbase/common/service", include: [
+		"UObject.java",
+		"JsonTokenStream.java",
+		"*Exception.java"
+	])
+
+	doLast {
+		delete "$tempunpack"
+	}
+}
+
+tasks.war {
```

Comment:
## Build the war file

Building the war file involves compiling the main classes, generating documentation, packaging everything into the appropriate directories, and then generating the war file from that.

### to test:
run `ant clean && ant compile && ant buildwar`
run `gradle expandWar` to expand the resultant file into its constituent parts, and rename the directory `war_contents` to something like `ant_war_contents`

run `gradle clean buildWar`
then expand the war file with `gradle expandWar`

The ant version of the war contains the test classes and dependencies.
```sh
❯ diff -r war_contents/WEB-INF/lib war_contents_ant/WEB-INF/lib
Only in war_contents_ant/WEB-INF/lib: arangodb-java-driver-6.7.2.jar
Only in war_contents_ant/WEB-INF/lib: byte-buddy-1.9.10.jar
Only in war_contents_ant/WEB-INF/lib: byte-buddy-agent-1.9.10.jar
Only in war_contents_ant/WEB-INF/lib: commons-collections-3.2.1.jar
Only in war_contents_ant/WEB-INF/lib: commons-lang-2.4.jar
Only in war_contents_ant/WEB-INF/lib: equalsverifier-3.1.10.jar
Only in war_contents_ant/WEB-INF/lib: java-semver-0.9.0.jar
Only in war_contents_ant/WEB-INF/lib: kbase-auth2test-0.2.4.jar
Only in war_contents_ant/WEB-INF/lib: mockito-core-3.0.0.jar
Only in war_contents_ant/WEB-INF/lib: objenesis-2.6.jar
Only in war_contents_ant/WEB-INF/lib: velocity-1.7.jar
Only in war_contents_ant/WEB-INF/lib: velocypack-2.4.0.jar
❯ diff -r war_contents/WEB-INF/classes/us war_contents_ant/WEB-INF/classes/us
Only in war_contents_ant/WEB-INF/classes/us/kbase/common: test
Only in war_contents_ant/WEB-INF/classes/us/kbase/typedobj/db: test
Only in war_contents_ant/WEB-INF/classes/us/kbase/typedobj: test
Only in war_contents_ant/WEB-INF/classes/us/kbase/workspace/listener: package-info.class
Only in war_contents_ant/WEB-INF/classes/us/kbase/workspace/modules: package-info.class
Only in war_contents_ant/WEB-INF/classes/us/kbase/workspace: test
```